kernel/object-store/src/main/java/io/mosip/commons/khazana/impl/S3Adapter.java

```diff
@@ -1,341 +1,465 @@
+        S3Object s3Object = null;
+        try {
+        	 String finalObjectName=null;
+        	 String bucketName=null;
+        	if(useAccountAsBucketname) {
+        		 finalObjectName = ObjectStoreUtil.getName(container,source, process, objectName);
+        		 bucketName=account;
+        	}else {
+        		 finalObjectName = ObjectStoreUtil.getName(source, process, objectName);
+        		 bucketName=container;
+        	}
+            ObjectMetadata objectMetadata = new ObjectMetadata();
+            //changed usermetadata getting  overrided
+            //metadata.entrySet().stream().forEach(m -> objectMetadata.addUserMetadata(m.getKey(), m.getValue() != null ? m.getValue().toString() : null));
+           
+            s3Object = getConnection(bucketName).getObject(bucketName, finalObjectName);
+            if (s3Object.getObjectMetadata() != null && s3Object.getObjectMetadata().getUserMetadata() != null)
+                s3Object.getObjectMetadata().getUserMetadata().entrySet().forEach(m -> objectMetadata.addUserMetadata(m.getKey(), m.getValue()));
+            metadata.entrySet().stream().forEach(m -> objectMetadata.addUserMetadata(m.getKey(), m.getValue() != null ? m.getValue().toString() : null));
+            PutObjectRequest putObjectRequest = new PutObjectRequest(bucketName, finalObjectName, s3Object.getObjectContent(), objectMetadata);
+            putObjectRequest.getRequestClientOptions().setReadLimit(readlimit);
+            getConnection(bucketName).putObject(putObjectRequest);
+            return metadata;
+        } catch (Exception e) {
+            LOGGER.error(SESSIONID, REGISTRATIONID,"Exception occured to addObjectMetaData for : " + container, ExceptionUtils.getStackTrace(e));
+            metadata = null;
+            throw new ObjectStoreAdapterException(OBJECT_STORE_NOT_ACCESSIBLE.getErrorCode(), OBJECT_STORE_NOT_ACCESSIBLE.getErrorMessage(), e);
+        } finally {
+            try {
+                if (s3Object != null)
+                    s3Object.close();
+            } catch (IOException e) {
+                LOGGER.error(SESSIONID, REGISTRATIONID,"IO occured : " + container, ExceptionUtils.getStackTrace(e));
+            }
+        }
+    }
+
+    @Override
+    public Map<String, Object> addObjectMetaData(String account, String container, String source, String process,
+                                                 String objectName, String key, String value) {
+        Map<String, Object> meta = new HashMap<>();
+        meta.put(key, value);
+        String finalObjectName=null;
+   	    
+   	   if(useAccountAsBucketname) {
+   		  finalObjectName = ObjectStoreUtil.getName(container,source, process, objectName);
+   		
+   	   }else {
+   		 finalObjectName = ObjectStoreUtil.getName(source, process, objectName);
+   		}
+       
+        return addObjectMetaData(account, container, source, process, finalObjectName, meta);

```

Comment:
This method is already concatinating the finalObjectName, but why are we doing it again in this method?

Reply:
In previous release also it is concatinating so i have kept as it is ,it should not affect existing flow